airflow/ui/src/components/FlexibleForm/FieldString.tsx

```diff
@@ -19,27 +19,43 @@
 import { Input } from "@chakra-ui/react";
 
 import type { FlexibleFormElementProps } from ".";
+import { paramPlaceholder, useParamStore } from "../TriggerDag/useParamStore";
 
-export const FieldString = ({ name, param }: FlexibleFormElementProps) => (
-  <>
-    <Input
-      defaultValue={String(param.value ?? "")}
-      id={`element_${name}`}
-      list={param.schema.examples ? `list_${name}` : undefined}
-      maxLength={param.schema.maxLength ?? undefined}
-      minLength={param.schema.minLength ?? undefined}
-      name={`element_${name}`}
-      placeholder={param.schema.examples ? "Start typing to see options." : undefined}
-      size="sm"
-    />
-    {param.schema.examples ? (
-      <datalist id={`list_${name}`}>
-        {param.schema.examples.map((example) => (
-          <option key={example} value={example}>
-            {example}
-          </option>
-        ))}
-      </datalist>
-    ) : undefined}
-  </>
-);
+export const FieldString = ({ name }: FlexibleFormElementProps) => {
+  const { paramsDict, setParamsDict } = useParamStore();
+  const param = paramsDict[name] ?? paramPlaceholder;
+  const handleChange = (value: string) => {
+    if (paramsDict[name]) {
+      paramsDict[name].value = value === "" ? undefined : value;
```

Comment:
Just tested and if the value is set to "undefined" then it is removed from the params dict. THis leads to the case that if form is submitted that the DAG default might be used... not `null`. Therefore - even if ESLint always recommends to use `undefined` still we need to set it to `null` in this case.
```suggestion
      paramsDict[name].value = value === "" ? null : value;
```

Reply:
Let's make sure to include a comment when we add `eslint-ignore` then